operator/controllers/lifecycle/keptnapp/controller.go

```diff
@@ -159,6 +159,9 @@ func (r *KeptnAppReconciler) createAppVersion(ctx context.Context, app *klcv1alp
 	}
 
 	appVersion := app.GenerateAppVersion(previousVersion, traceContextCarrier)
+	if len(appVersion.ObjectMeta.Name) > common.MaxVersionLength {
```

Comment:
if we do the check on the `objectMeta.name` of the already generated appVersion, we can check for the maximum length of K8s objects (253) here